utils/filesystem.go

```diff
@@ -183,5 +189,69 @@ func UnZip(zipFileName, fileDestination string) {
 			errors.CheckErr(err, 404, "")
 		}
 	}
-	log.Println("File extracted:", zipFileName)
+	fmt.Printf("Extracted file from '%s' to '%s'\n", filePath, destination)
+	return nil
+}
+
+// UnTar unpacks a tar.gz file to a destination
+func UnTar(pathToTarFile, destination string) error {
+	fileReader, err := read(pathToTarFile)
+	if err != nil {
+		return err
+	}
+	defer fileReader.Close()
+	gzipReader, err := gzip.NewReader(fileReader)
+	if err != nil {
+		return err
+	}
+	defer gzipReader.Close()
+	tarReader := tar.NewReader(gzipReader)
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			// end of tar archive
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+		target := filepath.Join(destination, header.Name)
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.MkdirAll(target, os.FileMode(header.Mode)); err != nil {
+				log.Fatal(err)
+			}
+		case tar.TypeReg:
+			fileToOverwrite, err := overwrite(target)
+			defer fileToOverwrite.Close()
+			if err != nil {
+				log.Fatal(err)
+			}
+			if _, err := io.Copy(fileToOverwrite, tarReader); err != nil {
+				log.Fatal(err)
+			}
+		default:
+			log.Printf("Can't: %c, %s\n", header.Typeflag, target)
+		}
+	}
+	return nil
+}
+
+func overwrite(filePath string) (*os.File, error) {
+	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_TRUNC, 0777)
+	if err != nil {
+		file, err = os.Create(filePath)
+		if err != nil {
+			return file, err
+		}
+	}
+	return file, nil
+}
+
+func read(filePath string) (*os.File, error) {
```

Comment:
It's probably okay because the parameter is filePath (and it's only local) but could consider a more descriptive name for this function? Would you also want to export this though in future?